post_info.php

```diff
@@ -29,6 +29,21 @@
         case 'surveys':
             // Get the id of the survey
             $survey_id = $REQUEST_data['survey_id'];
+            // If the user wants to set the attribute "viewable_by_others", call
+            // the function "handle_viewable_by_others" if the variable is valid
+            if (isset($REQUEST_data['viewable_by_others'])) {
+                $viewable_by_others = $REQUEST_data['viewable_by_others'];
+                // Check the validation of $viewable_by_others
+                if (
+                    $viewable_by_others != true &&
+                    $viewable_by_others != false
```

Comment:
It could also be a string. Let's be permissive. Look at https://stackoverflow.com/questions/4775294/parsing-a-string-into-a-boolean-value-in-php